lib/ansible/modules/windows/win_power_plan.ps1

```diff
@@ -55,6 +55,27 @@ If (! ($all_available_plans.ContainsKey($name)) )
 #If false, means plan is not active and we move down to enable
 #Since the results here are the same whether check mode or not, no specific handling is required
 #for check mode.
+
+If ([System.Environment]::OSVersion.Version -ge '10.0.17134')
+{
+    $plan = Get-WmiObject -Class win32_powerplan -Namespace root\cimv2\power -Filter "ElementName='$name'"
+    $regex = [regex]"{(.*?)}$"
+    $planGuid = $regex.Match($plan.instanceID.Tostring()).groups[1].value
+
+    If ( $all_available_plans.item($name) )
+    {
+        Exit-Json $result
+    }
+
+    Else
+    {
+        $result.changed = $true
```

Comment:
I believe that you would want to put `$result.changed = $true` only after a changing action, so below `powercfg`. This way it will only reflect it being changed if the action succeeds. As it is, if `powercfg` fails Ansible will still show that a change occurred.

Reply:
I could not reproduce a scenario when the action failed but it showed it as changed, but I've moved this below the powercfg command anyway, seems logical.